src/NuGet.Core/NuGet.Packaging.Core/PackageType.cs

```diff
@@ -22,11 +28,42 @@ public PackageType(string name, Version version)
             Name = name;
             Version = version;
         }
-
-        public static PackageType Default { get; } = new PackageType("Legacy", version: new Version(0, 0));
-
+        
         public string Name { get; }
 
         public Version Version { get; }
+
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as PackageType);
+        }
+
+        public bool Equals(PackageType other)
+        {
+            if (other == null)
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            return StringComparer.OrdinalIgnoreCase.Equals(Name, other.Name) && Version == other.Version;
+        }
+
+        public override int GetHashCode()
+        {
+            var combiner = new HashCodeCombiner();
+
+            combiner.AddObject(Name);
```

Comment:
Ported the implementation over from:
https://github.com/aspnet/Common/blob/dev/src/Microsoft.Extensions.HashCodeCombiner.Sources/HashCodeCombiner.cs#L72